server/graphql/types/src/types/program/vaccination_card.rs

```diff
@@ -87,6 +99,13 @@ impl VaccinationCardItemNode {
     pub async fn suggested_date(&self) -> &Option<NaiveDate> {
         &self.item.suggested_date
     }
+    pub async fn status(&self) -> &Option<VaccinationCardItemNodeStatus> {
+        &Some(VaccinationCardItemNodeStatus::Given)
+        // &self
+        //     .item
+        //     .status
+        //     .map(|status| VaccinationCardItemNodeStatus::from_domain(&status))
+    }
```

Comment:
@lache-melvin this is the line I can't quite get working. Commented out it what I'm trying to do. 

Compiler suggests:
```
cannot move out of `self.item.status` which is behind a shared reference
help: consider calling `.as_ref()` or `.as_mut()` to borrow the type's contents
```
But when I use `as_ref()`, the `#Object` macro above complains 🤷‍♂️

Reply:
Ah yeah, they're a bit pedantic 😅 This should work:

```rust
// remove & from return type, because the reference is created within this fn (the reference created by .as_ref() wouldn't live beyond the end of this fn call, so need to return the option for the caller to own)
pub async fn status(&self) -> Option<VaccinationCardItemNodeStatus> { 
  // so remove & here too
  self.item
    .status
    // take a reference of item status (could also clone, but more expensive)
    .as_ref()
    // status here is now &VaccinationCardItemStatus :)
    .map(|status| VaccinationCardItemNodeStatus::from_domain(status))
}
```